packages/webdriver/src/utils.js

```diff
@@ -198,11 +198,19 @@ export function isW3C (capabilities) {
      * assume session to be a WebDriver session when
      * - capabilities are returned
      *   (https://w3c.github.io/webdriver/#dfn-new-sessions)
-     * - a `platform` capability is not given but a platformName is returned
-     *   which is not defined in the JSONWire protocol
+     * - it is an Appium session (since Appium is full W3C compliant)
      */
     const isAppium = capabilities.automationName || capabilities.deviceName
-    return Boolean((!capabilities.platform && capabilities.platformName) || isAppium)
+    const hasW3CCaps = (
+        capabilities.platformName &&
+        capabilities.browserVersion &&
+        /**
+         * local safari doesn't provide platformVersion therefor
+         * check also for setWindowRect
+         */
+        (capabilities.platformVersion || capabilities.setWindowRect)
```

Comment:
One more question, does all w3c browsers have either platform version or setWindowRect?

Reply:
yes both caps are part of the W3C spec and __not__ within jsonwp. Usually `platformVersion` should exist in any session response but I don't see it only for running local Safari sessions which is why I added another check for `setWindowRect` which is also commonly set.

Reply:
this is definitely not a 100% bullet proof solution yet but improves the problem we have with Edge on Sauce.